WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -15,29 +15,196 @@ struct AddEditCoupon: View {
 
     var body: some View {
         NavigationView {
+            GeometryReader { geometry in
+                ScrollView {
+                    VStack (alignment: .leading) {
+                        Group {
+                            ListHeaderView(text: Localization.headerCouponDetails.uppercased(), alignment: .left)
 
-            //TODO: implement the content of the view
-            Text("Hello, World!")
-                .toolbar {
-                                    ToolbarItem(placement: .cancellationAction) {
-                                        Button("Cancel", action: {
-                                            presentation.wrappedValue.dismiss()
-                                        })
-                                    }
-                                }
+                            Group {
+                                TitleAndTextFieldRow(title: Localization.couponAmountPercentage,
+                                                     placeholder: Localization.couponAmountPercentage,
+                                                     text: $viewModel.amountField,
+                                                     editable: false,
+                                                     fieldAlignment: .leading,
+                                                     keyboardType: .asciiCapableNumberPad)
+                                Divider()
+                                    .padding(.leading, Constants.margin)
+                            }
+
+                            Text(Localization.footerCouponAmountPercentage)
+                                .subheadlineStyle()
+                                .padding(.horizontal, Constants.margin)
+
+                            Group {
+                                TitleAndTextFieldRow(title: Localization.couponCode,
+                                                     placeholder: Localization.couponCode,
+                                                     text: $viewModel.codeField,
+                                                     editable: false,
+                                                     fieldAlignment: .leading,
+                                                     keyboardType: .asciiCapableNumberPad)
```

Comment:
Same comment as above - please reconsider the placeholder, editable and keyboard type for the row. I'm thinking we only allow alpha-numeric characters, so maybe keyboard type `.namePhonePad` is more suitable?